tests/unit/test_conf.py

```diff
@@ -263,19 +261,98 @@ def test_init_session_config_with_not_valid_key():
     assert session_config.connection_acquisition_timeout == 333
 
 
-def test_pool_config_deprecated_update_routing_table_timeout():
-    with _pool_config_deprecations():
-        _ = PoolConfig.consume({"update_routing_table_timeout": 1})
-
-    with warnings.catch_warnings():
-        warnings.simplefilter("error")
-        _ = PoolConfig.consume({})
-
-
-def test_session_config_deprecated_session_connection_timeout():
-    with _session_config_deprecations():
-        _ = SessionConfig.consume({"session_connection_timeout": 1})
-
-    with warnings.catch_warnings():
-        warnings.simplefilter("error")
-        _ = SessionConfig.consume({})
+@pytest.mark.parametrize("config", (
+    {},
+    {"encrypted": False},
+    {"trust": TRUST_SYSTEM_CA_SIGNED_CERTIFICATES},
+    {"trust": TRUST_ALL_CERTIFICATES},
+))
+def test_no_ssl_mock(config, mocker):
+    ssl_context_mock = mocker.patch("ssl.SSLContext", autospec=True)
+    pool_config = PoolConfig.consume(config)
+    assert pool_config.encrypted is False
+    assert pool_config.get_ssl_context() is None
+    ssl_context_mock.assert_not_called()
+
+
+@pytest.mark.parametrize("config", (
+    {"encrypted": True},
+    {"encrypted": True, "trust": TRUST_SYSTEM_CA_SIGNED_CERTIFICATES},
+))
+def test_trust_system_cas_mock(config, mocker):
+    ssl_context_mock = mocker.patch("ssl.SSLContext", autospec=True)
+    pool_config = PoolConfig.consume(config)
+    assert pool_config.encrypted is True
+    ssl_context = pool_config.get_ssl_context()
+    _assert_mock_tls_1_2(ssl_context_mock)
+    ssl_context_mock.return_value.load_default_certs.assert_called_once_with()
+    ssl_context_mock.return_value.load_verify_locations.assert_not_called()
+    assert ssl_context.check_hostname
+    assert ssl_context.verify_mode
+
+
+@pytest.mark.parametrize("config", (
+    {"encrypted": True, "trust": TRUST_ALL_CERTIFICATES},
+))
+def test_trust_all_mock(config, mocker):
+    ssl_context_mock = mocker.patch("ssl.SSLContext", autospec=True)
+    pool_config = PoolConfig.consume(config)
+    assert pool_config.encrypted is True
+    ssl_context = pool_config.get_ssl_context()
+    _assert_mock_tls_1_2(ssl_context_mock)
+    ssl_context_mock.return_value.load_verify_locations.assert_not_called()
+    assert ssl_context.check_hostname is False
+    assert ssl_context.verify_mode is ssl.CERT_NONE
+
+
+def _assert_mock_tls_1_2(mock):
+    mock.assert_called_once_with(ssl.PROTOCOL_TLS_CLIENT)
+    if sys.version_info >= (3, 7):
+        assert mock.return_value.minimum_version == ssl.TLSVersion.TLSv1_2
+    else:
+        assert mock.mock_calls[1][1][0] == ssl.OP_NO_TLSv1
```

Comment:
1) I find `mock.mock_calls[1][1][0]` hard to understand
2) It's pretty fragile as it asserts that the seconds thing done with the SSLContext is t configuring `ssl.OP_NO_TLSv1`. What if we configure something else before?
3) This only tests if TLS1 is disabled. We also want to ensure that TLS1.1 is disabled.

Here's a suggestion on how it could be done. I tried doing it with mock call assertions but that didn't really pan out because the `__ior__` calls would stack up and you'd still have to assume some configuration order.
```suggestion
@pytest.mark.parametrize("config", (
    {"encrypted": True},
    {"encrypted": True, "trust": TRUST_SYSTEM_CA_SIGNED_CERTIFICATES},
))
def test_trust_system_cas_mock(config, mocker):
    ssl_context_mock = mocker.patch("ssl.SSLContext", autospec=True)
    ssl_context_mock.return_value.options = 0
    pool_config = PoolConfig.consume(config)
    assert pool_config.encrypted is True
    ssl_context = pool_config.get_ssl_context()
    _assert_mock_tls_1_2(ssl_context_mock)
    ssl_context_mock.return_value.load_default_certs.assert_called_once_with()
    ssl_context_mock.return_value.load_verify_locations.assert_not_called()
    assert ssl_context.check_hostname
    assert ssl_context.verify_mode


@pytest.mark.parametrize("config", (
    {"encrypted": True, "trust": TRUST_ALL_CERTIFICATES},
))
def test_trust_all_mock(config, mocker):
    ssl_context_mock = mocker.patch("ssl.SSLContext", autospec=True)
    ssl_context_mock.return_value.options = 0
    pool_config = PoolConfig.consume(config)
    assert pool_config.encrypted is True
    ssl_context = pool_config.get_ssl_context()
    _assert_mock_tls_1_2(ssl_context_mock)
    ssl_context_mock.return_value.load_verify_locations.assert_not_called()
    assert ssl_context.check_hostname is False
    assert ssl_context.verify_mode is ssl.CERT_NONE


def _assert_mock_tls_1_2(mock):
    mock.assert_called_once_with(ssl.PROTOCOL_TLS_CLIENT)
    if sys.version_info >= (3, 7):
        assert mock.return_value.minimum_version == ssl.TLSVersion.TLSv1_2
    else:
        assert mock.return_value.options & ssl.OP_NO_TLSv1
        assert mock.return_value.options & ssl.OP_NO_TLSv1_1
        assert not mock.return_value.options & ssl.OP_NO_TLSv1_2
        assert not mock.return_value.options & ssl.OP_NO_TLSv1_3
```